rules/Transform/Rector/ClassMethod/ReturnTypeWillChangeRector.php

```diff
@@ -92,27 +100,24 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        /** @var array<string, string[]> $classMethodsOfClass */
-        $classMethodsOfClass = array_merge_recursive($this->resolveDefaultConfig(), $this->classMethodsOfClass);
-        $className = (string) $this->nodeNameResolver->getName($classLike);
-        $objectType = new ObjectType($className);
-        $methodName = $this->nodeNameResolver->getName($node);
+        $classReflection = $this->reflectionResolver->resolveClassAndAnonymousClass($classLike);
+        $methodName = $node->name->toString();
+
         $hasChanged = false;
 
-        foreach ($classMethodsOfClass as $class => $methods) {
-            $configuredClassObjectType = new ObjectType($class);
-            if (! $configuredClassObjectType->isSuperTypeOf($objectType)->yes()) {
+        foreach ($this->returnTypeChangedClassMethodReferences as $returnTypeChangedClassMethodReference) {
+            if (! $classReflection->isSubclassOf($returnTypeChangedClassMethodReference->getClass())) {
```

Comment:
The `->getClass()` seems unneeded as no need to get the ClassReflection, just verify sub type

Reply:
Not sure what you mean. How would you rephrase the code?

Reply:
Oh, I see, I missed that it called against `$returnTypeChangedClassMethodReference`